gridstatus/isone.py

```diff
@@ -817,6 +811,51 @@ def _create_interval_start_from_hour_start(self, data):
 
         return data
 
+    def _filter_intervals_in_range(self, date, end, intervals):
+        """Filters intervals to those are at least partially within the given range.
+        If no end is provided, includes all intervals that start after the given date.
+
+        Args:
+            date (Datetime): Start datetime. Must be in self.default_timezone
+            end (Optional[Datetime]): End datetime. Must be in self.default_timezone
+                if provided
+            intervals (list): List of intervals in the format "HH-HH"
+        """
+        if (date.tz == timezone.utc) or (date.tz.zone != self.default_timezone):
+            raise ValueError("Date must be in default timezone")
+
+        if end:
+            if (end.tz == timezone.utc) or (end.tz.zone != self.default_timezone):
+                raise ValueError("End must be in default timezone")
+            if end.date() != date.date():
+                raise ValueError("Date and end must be on the same day")
+
+        # Convert intervals to hour ranges
+        interval_hours = [
+            (int(interval.split("-")[0]), int(interval.split("-")[1]))
+            for interval in intervals
+        ]
+
+        if not end:
+            end = date.replace(hour=23, minute=59, second=59, microsecond=0)
```

Comment:
If this method is only called on th current date, then I think this should be instead set to the current time.